src/main/java/ticTacToe/game/GameFlow.java

```diff
@@ -35,35 +34,24 @@ public void runGame() {
     }
 
     private void setUpPlayers() {
+        String opponentOptionNumber = ui.chooseOpponent();
         firstPlayer = playersFactory.firstPlayer(ui);
-        secondPlayer = playersFactory.secondPlayer(firstPlayer.getMark());
+        secondPlayer = playersFactory.secondPlayer(opponentOptionNumber, firstPlayer.getMark());
     }
 
     private void gameFlow() {
         String currentMark = ui.askForStarter();
-        while (isOnGoingGame()) {
+        while (!grid.isEndedGame(lines)) {
```

Comment:
Here and throughout the code you use `grid` and `lines` together. `lines` is rarely used on its own (I think once). That tells me that these two variables are buddies.
Is there any value in having them independent of each other? Isn't `Lines` a detail of the `Grid`?
Could you refactor the places that use `grid` and `lines` together to just use the grid?
Then you could shrink the number of arguments and fields  by one 😄

Reply:
Definitely this needs a refactoring...